lib/network.py

```diff
@@ -641,13 +635,15 @@ def cb2(x):
         return cb2
 
     def subscribe_to_addresses(self, addresses, callback):
-        hashes = [self.addr_to_scripthash(addr) for addr in addresses]
-        msgs = [('blockchain.scripthash.subscribe', [x]) for x in hashes]
-        self.send(msgs, self.overload_cb(callback))
+        hash2address = {bitcoin.address_to_scripthash(address): address for address in addresses}
+        self.h2addr.update(hash2address)
+        msgs = [('blockchain.scripthash.subscribe', [x]) for x in hash2address.keys()]
```

Comment:
note: as a result of this, the subscriptions for `addresses` will be in a different than before order (sorted) on the wire, but I believe this is OK.